percona/controller/pgbackup/controller.go

```diff
@@ -505,7 +470,7 @@ func updatePGBackrestInfo(ctx context.Context, c client.Client, pod *corev1.Pod,
 }
 
 func finishBackup(ctx context.Context, c client.Client, pgBackup *v2.PerconaPGBackup, job *batchv1.Job) (*reconcile.Result, error) {
-	if checkBackupJob(job) == v2.BackupSucceeded {
+	if job != nil && checkBackupJob(job) == v2.BackupSucceeded {
```

Comment:
Should we maybe validate the input job once at top of the function and avoid repeating the same check accross different places?

e.g.



```
func finishBackup(ctx context.Context, c client.Client, pgBackup *v2.PerconaPGBackup, job *batchv1.Job) (*reconcile.Result, error) {
	if job == nil {
		// do something
	}
```

Reply:
It will be clearer to repeat this check. We can't change the order of each action in this function, so if we add the `if job == nil` check at the top, it will just have a lot of duplicated code.